### ISSUE TITLE
Possibility configure listen port via Environment variable

### ISSUE THREAD

**Issue by abtris** [Caiyeon/goldfish#264]
### Feature request:

I can deploy app to Heroku, but you have to listen on `$PORT` that isn't easy possiblem without change in app.

Is this a front-end feature or backend? (or both?)

- Backend

Description of feature:

- Can override listen port from config via environment variable

Are you able to submit a pull request for this feature?

- yes, I hope.

**Comment by Caiyeon** [comment]
A simple `echo` might be useful in constructing a config file in the deployment script, and a `sed` command might be useful in substituting values in a templated file. I don't think you would be the first to construct a config file at deployment time.

In fact, terraform has a template system which translates environment variables to files just for this problem. Therefore, I would say that this is out of the scope of the goldfish program.

In addition, relating environment variables to configuration sets a precedent, and one that would require more documentation. For example, why would `PORT` be a translatable env variable, while `ADDR` is not? In addition, if `PORT` is set and the config file provides a different port, which one should take priority? Should a warning label be given because the second priority setting is suppressed? There are too many questions to solve by going this route.

In short, I believe this is a deployment-time problem, and the cons outweigh the pros.

**Comment by abtris** [comment]
There is concept https://12factor.net/config that I'm using for distributed apps for deploy that can override config using env variables. It's used in PaaS quite often. Deployment not using script just git push.

**Comment by Caiyeon** [comment]
On the other hand, vault explicitly requires a config file to launch https://www.vaultproject.io/docs/configuration/index.html, and will not accept env variables in deployment (although it does as a CLI)

If you have deployed vault, goldfish is no different.
